scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/S3ConnectorUploader.java

```diff
@@ -0,0 +1,322 @@
+        try {
+            MultipleFileUpload uploader = transferManager.uploadFileList(bucketName, keyPrefix, new File("."), files);
+            // loop with Transfer.isDone()
+            SchedulerExamplesUtils.showTransferProgress(uploader);
+            // or block with Transfer.waitForCompletion()
+            SchedulerExamplesUtils.waitForCompletion(uploader);
+        } catch (AmazonServiceException e) {
+            getErr().println(e.getErrorMessage());
+            System.exit(1);
+        }
+        transferManager.shutdownNow();
+    }
+
+    /**
+     * Upload a local directory to S3. <br>
+     * Requires a bucket name. <br>
+     *
+     * @param filePath
+     * @param bucketName
+     * @param keyPrefix
+     * @param pause
+     * @param s3Client
+     */
+    private void uploadFile(String filePath, String bucketName, String keyPrefix, boolean pause, AmazonS3 s3Client) {
+        getOut().println("file: " + filePath + (pause ? " (" + PAUSE + ")" : ""));
+
+        File file = new File(filePath);
+
+        String keyName = (keyPrefix != null) ? Paths.get(keyPrefix, file.getName()).toString() : file.getName();
+
+        TransferManager transferManager = TransferManagerBuilder.standard().withS3Client(s3Client).build();
+        try {
+            Upload uploader = transferManager.upload(bucketName, keyName, file);
+            // loop with Transfer.isDone()
+            SchedulerExamplesUtils.showTransferProgress(uploader);
+            //  or block with Transfer.waitForCompletion()
+            SchedulerExamplesUtils.waitForCompletion(uploader);
+        } catch (AmazonServiceException e) {
+            getErr().println(e.getErrorMessage());
+            System.exit(1);
+        }
+        transferManager.shutdownNow();
+    }
+
+    /**
+     * Get or initialize the S3 client.
+     * Note: this method must be synchronized because we're accessing the
+     * field and we're calling this method from a worker thread.
+     *
+     * @return the S3 client
+     */
+    private synchronized AmazonS3 getS3Client(String region) {
+
+        BasicAWSCredentials credentials = new BasicAWSCredentials(accessKey, secretKey);
+
+        AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard()
+                                                             .withCredentials(new AWSStaticCredentialsProvider(credentials));
+        builder = builder.withRegion(region);
+        builder = builder.withPathStyleAccessEnabled(pathStyle);
+        return builder.build();
+    }
+
+    /**
+     * Check if an S3 bucket exists and returns its name if it does exist or null otherwise.
+     * @param bucketName
+     * @param s3
+     * @return bucket name if it exists or null otherwise
+     */
+    private Bucket getBucket(String bucketName, AmazonS3 s3) {
+        Bucket namedBucket = null;
+        List<Bucket> buckets = s3.listBuckets();
+        for (Bucket b : buckets) {
```

Comment:
You can clean up the whole method like this : 
return s3.listBuckets().stream().filter(b -> b.getName().equals(bucketName)).findFirst().get();

Also the way it is implemented now, it s risky and does not perform well...imagine if there are 100000 buckets...even after finding the one u r searching for it will still loop through all the others